pkg/controller/server_status_request_controller_test.go

```diff
@@ -250,8 +95,152 @@ var _ = Describe("Server Status Request Reconciler", func() {
+					{
+						Name: "custom.io/myown",
+						Kind: "VolumeSnapshotter",
+					},
+				}).
+				Result(),
+			reqPluginLister: &fakePluginLister{
+				plugins: []framework.PluginIdentifier{
+					{
+						Name: "custom.io/myown",
+						Kind: "VolumeSnapshotter",
+					},
+				},
+			},
+			expected: statusRequestBuilder("1").
+				ServerVersion(buildinfo.Version).
+				Phase(velerov1api.ServerStatusRequestPhaseProcessed).
+				ProcessedTimestamp(now).
+				Plugins([]velerov1api.PluginInfo{
+					{
+						Name: "custom.io/myown",
+						Kind: "VolumeSnapshotter",
+					},
+				}).
+				Result(),
+			expectedRequeue: ctrl.Result{Requeue: false, RequeueAfter: statusRequestResyncPeriod},
+		}),
+		Entry("with phase=new will be processed and phased successfully patched", request{
+			req: statusRequestBuilder("1").
+				ServerVersion(buildinfo.Version).
+				Phase(velerov1api.ServerStatusRequestPhaseNew).
+				ProcessedTimestamp(now).
+				Plugins([]velerov1api.PluginInfo{
+					{
+						Name: "custom.io/myown",
+						Kind: "VolumeSnapshotter",
+					},
+				}).
+				Result(),
+			reqPluginLister: &fakePluginLister{
+				plugins: []framework.PluginIdentifier{
+					{
+						Name: "custom.io/myown",
+						Kind: "VolumeSnapshotter",
+					},
+				},
+			},
+			expected: statusRequestBuilder("1").
+				ServerVersion(buildinfo.Version).
+				Phase(velerov1api.ServerStatusRequestPhaseProcessed).
+				ProcessedTimestamp(now).
+				Plugins([]velerov1api.PluginInfo{
+					{
+						Name: "custom.io/myown",
+						Kind: "VolumeSnapshotter",
+					},
+				}).
+				Result(),
+			expectedRequeue: ctrl.Result{Requeue: false, RequeueAfter: statusRequestResyncPeriod},
+		}),
+		Entry("with phase=Processed does not get deleted if not expired", request{
+			req: statusRequestBuilder("1").
+				ServerVersion(buildinfo.Version).
+				Phase(velerov1api.ServerStatusRequestPhaseProcessed).
+				ProcessedTimestamp(now). // not yet expired
+				Plugins([]velerov1api.PluginInfo{
+					{
+						Name: "custom.io/myotherown",
+						Kind: "VolumeSnapshotter",
+					},
+				}).
+				Result(),
+			reqPluginLister: &fakePluginLister{
+				plugins: []framework.PluginIdentifier{
+					{
+						Name: "custom.io/myotherown",
+						Kind: "VolumeSnapshotter",
+					},
+				},
+			},
+			expected: statusRequestBuilder("1").
+				ServerVersion(buildinfo.Version).
+				Phase(velerov1api.ServerStatusRequestPhaseProcessed).
+				ProcessedTimestamp(now).
+				Plugins([]velerov1api.PluginInfo{
+					{
+						Name: "custom.io/myown",
+						Kind: "VolumeSnapshotter",
+					},
+				}).
+				Result(),
+			expectedRequeue: ctrl.Result{Requeue: false, RequeueAfter: statusRequestResyncPeriod},
+		}),
+		Entry("with phase=Processed gets deleted if expired", request{
+			req: statusRequestBuilder("1").
+				ServerVersion(buildinfo.Version).
+				Phase(velerov1api.ServerStatusRequestPhaseProcessed).
+				ProcessedTimestamp(now.Add(-61 * time.Second)). // expired
```

Comment:
nit: it would be good if we could calculate the expired time by referencing the `ttl` https://github.com/vmware-tanzu/velero/blob/e69fac1/pkg/controller/server_status_request_controller.go#L37, but I won't block this PR.